lib/spack/docs/packaging_guide.rst

```diff
@@ -853,66 +880,72 @@ Default branch
   .. code-block:: python
 
      class Example(Package):
-         ...
-         version('develop', git='https://github.com/example-project/example.git')
 
-  This download method is untrusted, and is not recommended.
+         git = "https://github.com/example-project/example.git"
 
-Tags
-  To fetch from a particular tag, use the ``tag`` parameter along with
-  ``git``:
+         version('develop')
+
+  This download method is untrusted, and is not recommended. Aside from HTTPS,
+  there is no way to verify that the repository has not been compromised, and
+  the commit you get when you install the package likely won't be the same
+  commit that was used when the package was first written. Additionally, the
+  default branch may change. It is best to at least specify a branch name.
```

Comment:
I tried to expand on each of these "This download method is untrusted, and is not recommended" statements to explain the rationale and what alternatives are available. I think this is sufficient to close #6785.